From 66f91c2ed6372c2688cd7f1ed3b82f7af3a2f18e Mon Sep 17 00:00:00 2001
From: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Date: Wed, 1 Jul 2020 04:47:26 +0530
Subject: [PATCH 1405/1851] spi: spi-xilinx: Check return value of
 of_property_read_u32

commit 27f06c117f62a57ebf4977c4f36dccb06ee5274a from
https://github.com/Xilinx/linux-xlnx.git

In two different instances the return value of "of_property_read_u32" API
was neither captured nor checked. Fixed it by capturing the return value
and then checking for any error.

Addresses-Coverity: "check_return"
Signed-off-by: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Acked-by: Naga Sureshkumar Relli <naga.sureshkumar.relli@xilinx.com>
Signed-off-by: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/spi/spi-xilinx.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/drivers/spi/spi-xilinx.c b/drivers/spi/spi-xilinx.c
index ed1005c00ddd..9acfb6a65078 100644
--- a/drivers/spi/spi-xilinx.c
+++ b/drivers/spi/spi-xilinx.c
@@ -649,8 +649,9 @@ static int xilinx_spi_probe(struct platform_device *pdev)
 	u32 tmp, rx_bus_width, fifo_size;
 	bool startup_block;
 
-	of_property_read_u32(pdev->dev.of_node, "num-cs",
-				&num_cs);
+	if (of_property_read_u32(pdev->dev.of_node, "num-cs", &num_cs))
+		dev_info(&pdev->dev,
+			 "Missing num-cs optional property, assuming default as <1>\n");
 	if (!num_cs)
 		num_cs = 1;
 
@@ -681,8 +682,10 @@ static int xilinx_spi_probe(struct platform_device *pdev)
 			"Missing fifo size\n");
 		return -EINVAL;
 	}
-	of_property_read_u32(pdev->dev.of_node, "bits-per-word",
-			     &bits_per_word);
+	if (of_property_read_u32(pdev->dev.of_node, "bits-per-word",
+				 &bits_per_word))
+		dev_info(&pdev->dev,
+			 "Missing bits-per-word optional property, assuming default as <8>\n");
 
 	xspi->rx_bus_width = XSPI_ONE_BITS_PER_WORD;
 	for_each_available_child_of_node(pdev->dev.of_node, nc) {
-- 
2.31.1

